fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/ExtraModelKey.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.model.loading.v1;
+
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.Contract;
+
+/**
+ * A unique key representing an extra model, not tied to a blockstate or item model.
+ *
+ * <p>Extra models can be registered with a {@link ModelLoadingPlugin} (see
+ * {@link ModelLoadingPlugin.Context#addModel(ExtraModelKey, UnbakedExtraModel)}). Once baking is complete, they may
+ * then be queried from the model manager using {@link FabricBakedModelManager#getModel(ExtraModelKey)}.
+ *
+ * @param <T> The type of the baked model.
+ * @see FabricBakedModelManager#getModel(ExtraModelKey)
+ * @see ModelLoadingPlugin.Context#addModel(ExtraModelKey, UnbakedExtraModel)
+ */
+public final class ExtraModelKey<T> {
```

Comment:
Am I missing something? Where is `name` used other than the .toString()? `ExtraModelKey` seems to only be used in a hashmap but never overrides hashcode?

Would this class be better as an interface an an impl record?

Reply:
`ExtraModelKey` is supposed to always have identity equivalence, and the name is only for debug purposes. Turning it into an interface (extendable) allows users to violate that contract, turning it into an interface (non-extendable) with a record impl would automatically violate that contract, and turning it into an interface (non-extendable) with a final class impl would be unnecessary abstraction for what is a mostly empty class.